### ISSUE TITLE
fio "you need to specify size=" error message

### ISSUE THREAD

**Issue by Gabson1** [axboe/fio#1585]
**Please acknowledge the following before creating a ticket**

- [X] I have read the GitHub issues section of [REPORTING-BUGS](../blob/master/REPORTING-BUGS).

**Description of the bug:**
Here is the full output...
```
========================================
Running benchmark test via "iops-config"
=========================================

performance: (g=0): rw=read, bs=(R) 4096B-4096B, (W) 4096B-4096B, (T) 4096B-4096B, ioengine=psync, iodepth=1
fio-3.25
Starting 1 process
Run status group 0 (all jobs):

========================================
Benchmark tests have concluded, logs have been written to "/fio/results"
=========================================

performance: you need to specify size=
fio: pid=0, err=22/file:filesetup.c:1055, func=total_file_size, error=Invalid argument
```

**Environment**: 
* local: macOS 12.5.1
* remote: K8 cluster v1.24.12

**fio version**: 3.25

**Reproduction steps**
Gist with the relevant configuration for reproduction: https://gist.github.com/Gabson1/5d07bd5ed1f749eca521c97b0fe6e2ad

Basically I am trying to test the performance of the CephCluster (rook.io) using fio.
I am new to using fio so I am not quite sure what I am doing wrong. I have tried looking for an answer on my own, but could not find a fix yet. 

I would greatly appreciate any type of help. Thank you in advance!

**Comment by mcluseau** [comment]
Hi, FWIW that many years later, I've seen that when there's data on the device, ie:
```
# fio [...]
fio-3.33
Starting 1 process
ceph-iops: you need to specify size=
# dd if=/dev/zero of=/dev/disk/... bs=1M count=100
# fio [...]
Starting 1 process
Jobs: 1 (f=1): [W(1)][50.9%][w=882MiB/s][w=226k IOPS][eta 07m:22s]
```

**Comment by sitsofe** [comment]
@Gabson1 apologies for the very late reply. Looking at the iops-config.fio job file in the gist you linked:
```
[global]
bs=4K
iodepth=256
direct=0
ioengine=libaio
group_reporting
time_based
runtime=900
numjobs=4
rw=randrw
size=50%
rwmixread=60
rwmixwrite=40
```

If the file doesn't already exist before fio starts, then [`size`](https://fio.readthedocs.io/en/latest/fio_doc.html#cmdoption-arg-size) has to be a fixed number i.e. you can only use percentages with `size` with existing files whose size can be looked up and from which the percentage of can be calculated.

@mcluseau That looks suspicious but you didn't include your full job file so it's hard to say much... Are you sure that the device in `/dev/disk/` you thought you were using exists e.g. if you run `file` on it does it say it is a device of a plain file? You may find that your `dd` created a new 100 megabyte file inside `/dev/` (which is presumably a udev filesystem). Can you let us know if that's the case?

**Comment by mcluseau** [comment]
hi @sitsofe, sorry I didn't mean to resurrect something, just bringing an potential thing to look at.
Replacing previous comment as, well, yes, the device link got removed concurrently as a result of another admin's action.
Still, thanks for the great software :)

**Comment by sitsofe** [comment]
@mcluseau:

Thanks for following up. I had a longer reply to your original comment but you've preempted it and I'm glad you've found fio useful - it's great to hear kind words from time to time :-) 

If we don't hear back from the original reporter I think we can close this issue and let the comments stand in case others hit similar issues.

**Comment by mcluseau** [comment]
damn, what were the odds of a blkdiscard on the full disk removing the partuuid symlink could happen as I was editing the command??? 😅 

I think a *lot* of cephers find fio useful!* You'll find it stated 12 times here for instance https://github.com/TheJJ/ceph-cheatsheet :)

I agree with you issue management.

Cheers!

(*) edit: also, here, it shows that writing to tmpfs was going the same speed as writing on the nvme drive, so the unexpected bottleneck is VM memory speed (or syscall rate meaning CPU bound?), not the drive :o

**Comment by sitsofe** [comment]
Closing due to lack of reply from the original reporter. If this issue is still happening with the latest fio (see https://github.com/axboe/fio/releases to find out which version that is) please reopen. Thanks!

**Comment by Gabson1** [comment]
Ah sorry for not replying appropriately, it's a bit ago and I don't remember what my fix was... but I'm pretty sure @sitsofe input was what fixed my "problem".

Great software! Thanks for your hard work
